luigi/parameter.py

```diff
@@ -995,3 +995,49 @@ def parse(self, s):
                 raise OSError("could not make parent directories for {s}".format(
                     s=s))
         return s
+
+
+class ChoiceParameter(Parameter):
+    """
+    A parameter which takes two values:
+        1. an instance of :class:`~collections.Iterable` and
+        2. the class of the variables to convert to.
+
+    In the task definition, use
+
+    .. code-block:: python
+
+        class MyTask(luigi.Task):
+            my_param = luigi.ChoiceParameter(choices=[0.1, 0.2, 0.3], var_type=float)
+
+    At the command line, use
+
+    .. code-block:: console
+
+        $ luigi --module my_tasks MyTask --my-param 0.1
+
+    Consider using :class:`~luigi.EnumParameter` for a typed, structured
+    alternative.  This class can perform the same role when all choices are the
+    same type and transparency of parameter value on the command line is
+    desired.
+    """
+    def __init__(self, var_type=str, *args, **kwargs):
+        if "choices" not in kwargs:
+            raise ParameterException("A choices iterable must be specified")
+        self._choices = set(var_type(choice) for choice in kwargs.pop("choices"))
```

Comment:
As I pointed out [earlier](https://github.com/spotify/luigi/pull/1800#discussion_r73090575), I probably would prefer if we could somehow would assert instead of convert. But I'm not sure really what's best :)